src/coreclr/src/debug/createdump/main.cpp

```diff
@@ -43,59 +59,91 @@ int __cdecl main(const int argc, const char* argv[])
+                                               MiniDumpWithThreadInfo |
+                                               MiniDumpWithTokenInformation);
             }
             else if ((strcmp(*argv, "-t") == 0) || (strcmp(*argv, "--triage") == 0))
             {
-                minidumpType = MiniDumpFilterTriage;
+                minidumpType = (MINIDUMP_TYPE)(MiniDumpFilterTriage |
+                                               MiniDumpWithThreadInfo);
             }
             else if ((strcmp(*argv, "-u") == 0) || (strcmp(*argv, "--full") == 0))
             {
-                minidumpType = MiniDumpWithFullMemory;
+                minidumpType = (MINIDUMP_TYPE)(MiniDumpWithFullMemory |
+                                               MiniDumpWithDataSegs |
+                                               MiniDumpWithHandleData |
+                                               MiniDumpWithUnloadedModules |
+                                               MiniDumpWithFullMemoryInfo |
+                                               MiniDumpWithThreadInfo |
+                                               MiniDumpWithTokenInformation);
             }
             else if ((strcmp(*argv, "-d") == 0) || (strcmp(*argv, "--diag") == 0))
             {
                 g_diagnostics = true;
             }
             else {
-                pid = atoll(*argv);
+                pid = atoi(*argv);
             }
             argv++;
         }
     }
 
     if (pid != 0)
     {
+        ArrayHolder<char> tmpPath = new char[MAX_LONGPATH];
+        ArrayHolder<char> dumpPath = new char[MAX_LONGPATH];
+
         if (dumpPathTemplate == nullptr)
         {
-            char tmpPath[MAX_LONGPATH];
             if (::GetTempPathA(MAX_LONGPATH, tmpPath) == 0)
             {
                 fprintf(stderr, "GetTempPath failed (0x%08x)", ::GetLastError());
                 return ::GetLastError();
             }
-            exitCode = strcat_s(tmpPath, MAX_LONGPATH, "coredump.%d");
+            exitCode = strcat_s(tmpPath, MAX_LONGPATH, DEFAULT_DUMP_TEMPLATE);
             if (exitCode != 0)
             {
                 fprintf(stderr, "strcat_s failed (%d)", exitCode);
                 return exitCode;
             }
             dumpPathTemplate = tmpPath;
         }
-        ReleaseHolder<DumpDataTarget> dataTarget = new DumpDataTarget(pid);
-        ReleaseHolder<CrashInfo> crashInfo = new CrashInfo(pid, dataTarget, false);
 
-        // The initialize the data target's ReadVirtual support (opens /proc/$pid/mem)
-        if (dataTarget->Initialize(crashInfo))
+        snprintf(dumpPath, MAX_LONGPATH, dumpPathTemplate, pid);
+
+        const char* dumpType = "minidump";
+        switch (minidumpType)
         {
-            if (!CreateDumpCommon(dumpPathTemplate, minidumpType, crashInfo))
-            {
-                exitCode = -1;
-            }
+            case MiniDumpWithPrivateReadWriteMemory:
```

Comment:
Given the values assigned above, can these cases even be reached? It seems the assigned type is always combination of several flags.

Reply:
Yes, you are right, the minidumpType switch should be masked with the bit in the case statements. I'll try to get that fixes in a future PR.